File: AnalyszeRunTimeResults.m
%% Setting Enviorment Parameters

AnalysisInitScript;

figureCounterSpec   = '%04d';

generateImages=1;

%% Loading Data

tRunTimeMatlab = readtable(fullfile('RunTimeData\', 'RunTimeMatlabTable.csv'));
mRunTimeMatlab=table2array(tRunTimeMatlab(2:end,2:end));
vMatrixSizeMatlab=table2array(tRunTimeMatlab(1,2:end));
sFunNameMatlab=table2array(tRunTimeMatlab(2:end,1));

tRunTimeJuliamkl = readtable(fullfile('RunTimeData\', 'RunTimeJuliamklTable.csv'));
mRunTimeJuliamkl=table2array(tRunTimeJuliamkl(2:end,2:end));
vMatrixSizeJuliamkl=table2array(tRunTimeJuliamkl(1,2:end));
sFunNameJuliamkl=table2array(tRunTimeJuliamkl(2:end,1));

tRunTimeJuliamklSIMD = readtable(fullfile('RunTimeData\', 'RunTimeJuliamklSIMDTable.csv'));
mRunTimeJuliamklSIMD=table2array(tRunTimeJuliamklSIMD(2:end,2:end));
vMatrixSizeJuliamklSIMD=table2array(tRunTimeJuliamklSIMD(1,2:end));
sFunNameJuliamklSIMD=table2array(tRunTimeJuliamklSIMD(2:end,1));

%% Displaying Results
figureIdx           = 0;

for ii = 1:size(mRunTimeMatlab,1)

    figureIdx   = figureIdx + 1;
    hFigure     = figure('Position', figPosMedium);
    hAxes       = axes();

    loglog(vMatrixSizeMatlab,mRunTimeMatlab(ii,:),'-o','LineWidth',lineWidthThin,'MarkerFaceColor','b'); hold on;
    loglog(vMatrixSizeJuliamkl,mRunTimeJuliamkl(ii,:),'-s','LineWidth',lineWidthThin,'MarkerFaceColor','r'); hold on;
    plotJuliaSIMD=ismember( sFunNameMatlab{ii}, sFunNameJuliamklSIMD ); % if 1 will plot JuliaSIMD
    if any(plotJuliaSIMD)
        h=loglog(vMatrixSizeJuliamklSIMD,mRunTimeJuliamklSIMD(plotJuliaSIMD,:),'-*','LineWidth',lineWidthThin,'MarkerFaceColor','y');
        legend('MATLAB','Julia-MKL','Julia-MKL-SIMD','Location','southeast')
    else
        legend('MATLAB','Julia-MKL','Location','southeast')
    end
    hold off;
    title(num2str(sFunNameMatlab{ii}));
    xlabel('Matrix Size');
    ylabel('Run Time  [micro Seconds]');

    if(generateImages == 1)
        set(hAxes, 'LooseInset', [0.05, 0.05, 0.05, 0.05]);
        saveas(hFigure,['Figures\Figure', num2str(figureIdx, figureCounterSpec), '.png']);
    end

end


%% Restoring Defaults

% set(0, 'DefaultFigureWindowStyle', 'normal');
% set(0, 'DefaultAxesLooseInset', defaultLooseInset);

File: AnalyszeRunTimeResults.jl
# using Pkg
# Pkg.add("Plots")
# Pkg.add("GR")

using DelimitedFiles
using LinearAlgebra
using Plots
# Setting Enviorment Parameters
generateImages=1;

# Loading Data
tRunTimeMatlab = readdlm("RunTimeData\\RunTimeMatlabTable.csv", ',');
mRunTimeMatlab=tRunTimeMatlab[2:end,2:end];
vMatrixSizeMatlab=tRunTimeMatlab[1,2:end];
sFunNameMatlab=tRunTimeMatlab[2:end,1];

tRunTimeJuliamkl = readdlm("RunTimeData\\RunTimeJuliamklTable.csv", ',');
mRunTimeJuliamkl=tRunTimeJuliamkl[2:end,2:end];
vMatrixSizeJuliamkl=tRunTimeJuliamkl[1,2:end];
sFunNameJuliamkl=tRunTimeJuliamkl[2:end,1];

tRunTimeJuliamklSIMD = readdlm("RunTimeData\\RunTimeJuliamklSIMDTable.csv", ',');
mRunTimeJuliamklSIMD=tRunTimeJuliamklSIMD[2:end,2:end];
vMatrixSizeJuliamklSIMD=tRunTimeJuliamklSIMD[1,2:end];
sFunNameJuliamklSIMD=tRunTimeJuliamklSIMD[2:end,1];

# Displaying Results
for ii = 1:size(sFunNameMatlab,2)

    plot( vMatrixSizeMatlab,[ mRunTimeMatlab[ii,:], mRunTimeJuliamkl[ii,:] ],
    labels=["MATLAB" "Julia-MKL"],
    xlabel="Matrix Size", ylabel="Run Time  [micro Seconds]",
    title="$(sFunNameMatlab[ii])", xaxis=:log, yaxis=:log );

    plotJuliaSIMD=ismember( fun, sFunNameJuliamklSIMD ); # if 1 will plot JuliaSIMD
    if any(plotJuliaSIMD)
        plot!(vMatrixSizeJuliamklSIMD,mRunTimeJuliamklSIMD[plotJuliaSIMD,:],
        label="Julia-MKL-SIMD",
        xaxis=:log, yaxis=:log,);
    end

    if(generateImages == 1)
        savefig("Figure$ii.png");
    end

end
